ansible/playbooks/roles/upgrade/tasks/grafana.yml

```
@@ -25,4 +29,4 @@
         name: grafana
         tasks_from: install
       when:
-        - grafana_defaults.grafana_version is version(ansible_facts.packages['grafana'][0].version, '>')
+        - grafana_version is version(ansible_facts.packages['grafana'][0].version, '>')
```

```suggestion
        - grafana_target_version is version(grafana_installed_version, '>')
```
I would also introduce upgrade state file, like we have here:
https://github.com/epiphany-platform/epiphany/blob/9fd3e99b50bf73920480798b8bd336688f3746a3/ansible/playbooks/roles/upgrade/tasks/opendistro_for_elasticsearch-01.yml#L35-L37